### ISSUE TITLE
images are not showing on all feed

### ISSUE THREAD

**Issue by Anmartawfik** [jsanahuja/InstagramFeed#61]
Looks like images are not showing in any feed , seems like same CORS issue

Many thanks
A

**Comment by nguyenhc-bitvn** [comment]
Same issue. Waiting for fixing...

**Comment by SpbSprut** [comment]
Same issue. Even examples on plugin's site don't show images but only texts.
https://www.sowecms.com/demos/InstagramFeed/#examples
https://prnt.sc/124z128

**Comment by ghost** [comment]
@jsanahuja, any suggestions about this issue? Attempts to access images via direct links return 403 with "URL signature expired" message:

403:
![image](https://user-images.githubusercontent.com/34391398/116242850-01e62300-a76f-11eb-82a2-0e84df57963c.png)

The message:
![image](https://user-images.githubusercontent.com/34391398/116242960-1aeed400-a76f-11eb-9752-be7b814e9286.png)

Also as far as I concerned, the issue with CORS and image access can be reproduced while your computer connects to some internet-endpoints (e.g. Wi-Fi).

**Comment by A1exMoody** [comment]
Yep..  Look like this 
![image](https://user-images.githubusercontent.com/62880209/116324154-3fc36580-a7c8-11eb-90ff-260d6a26b6c6.png)
The problem is also shown in the examples

**Comment by xchopin** [comment]
ugh I thought the problem was due to my apache cache thing; good to know I am not alone

**Comment by Nicolas-Cargo** [comment]
I hope @jsanahuja will come soon :/ I'm using it on so many websites..

**Comment by A1exMoody** [comment]
Me too...
I have to pass the order, but it all comes down to Instagram. I don't want used Insta API...

**Comment by CobeSean** [comment]
Adding myself on here, to follow for (hopefully) a solution soon...

**Comment by NosovN** [comment]
@jsanahuja, help us)

**Comment by Eugene1984** [comment]
@jsanahuja Adding myself here.

**Comment by igoluciano** [comment]
@jsanahuja Adding myself here.

**Comment by NoOneDesign** [comment]
same problem on all website where i use it. @jsanahuja if you fix it give me your paypal, you need a donation for this.

**Comment by nrayann** [comment]
@jsanahuja is the repo alive?

**Comment by evaleries** [comment]
If you have Cloudflare account you can fix this issue using Cloudflare workers by following these steps
https://gist.github.com/restyler/6c51e3ad20d7596e799d76e87cf93236

I've tried and it works.

**Comment by vgavrilovikj** [comment]
Was someone able to fix this issue..?

**Comment by ghost** [comment]
As mentioned [here](https://github.com/jsanahuja/jquery.instagramFeed/issues/134#issuecomment-826123273) we can use rapidapi as proxy. But there're limitations: 100 requests/month with free plan. There're a bunch of API's, but they have even harder limit: 20 requests/month or 20 requests/day.

**Comment by ribeiroeder** [comment]
Hello, not wanting to be negative, but I believe that it will be very unlikely to find a solution since instagram is very strict in unauthenticated access.

As I have several sites that depend on a smilar script, I searched and improved two scripts that use the new official method via the graph.instagram API.

The first uses javascript and the second PHP cURL, I am still testing it but the results have been great.

**Comment by ribeiroeder** [comment]
jQuery Ajax for feed Instagram Graph API
https://github.com/ribeiroeder/jquery-feed-instagram-graph

PHP cURL for feed Instagram Graph API
https://github.com/ribeiroeder/php-curl-instagram-graph

@jsanahuja Sorry for posting another third party solution here, your script is incredible, much more complex and well developed than the scripts above and it helped me for a long time. But either way I need to be generating alternatives because customers keep asking about a solution :(

**Comment by A1exMoody** [comment]
> If you have Cloudflare account you can fix this issue using Cloudflare workers by following these steps
> https://gist.github.com/restyler/6c51e3ad20d7596e799d76e87cf93236
> 
> I've tried and it works.

This is a good solution to the problem, but what if I have a hosting that is already optimized for everything you need, and I don't need to use Cloudflare?

**Comment by evaleries** [comment]
> > If you have Cloudflare account you can fix this issue using Cloudflare workers by following these steps
> > https://gist.github.com/restyler/6c51e3ad20d7596e799d76e87cf93236
> > I've tried and it works.
> 
> This is a good solution to the problem, but what if I have a hosting that is already optimized for everything you need, and I don't need to use Cloudflare?

Good question. You can create Cloudflare workers for free (100k requests/daily) and then you need modify a little bit of the library (See #62).
Let's say you have created Cloudflare workers, you can use my [modified version of this library](https://github.com/evaleries/InstagramFeed/blob/master/dist/InstagramFeed.min.js). Then, you need customize the instance of InstagramFeed a little bit, e.g
```javascript
new InstagramFeed({
    'username': 'travisci',
    'container': document.getElementById("instagram-feed5"),
    'display_profile': true,
    'cdnUrlResolver': function(url) {
        return 'https://your-cloudflare.workers.dev/' + url;
    }
});
```

Check my live demo:
[Main Site](https://pemro.id) - Check __Latest Instagram__ section
https://pemro.id/plugins/insta-feed/insta-feed.js
https://pemro.id/plugins/insta-feed/InstagramFeed.min.js - This uses my [modified version of this library](https://github.com/evaleries/InstagramFeed/blob/master/dist/InstagramFeed.min.js)

**Comment by A1exMoody** [comment]
> > > Если у вас есть учетная запись Cloudflare, вы можете решить эту проблему с помощью работников Cloudflare, выполнив следующие действия: 
> > > https://gist.github.com/restyler/6c51e3ad20d7596e799d76e87cf93236 
> > > Я пробовал, и он работает.
> > 
> > 
> > Это хорошее решение проблемы, но что, если у меня есть хостинг, который уже оптимизирован для всего, что вам нужно, и мне не нужно использовать Cloudflare?
> 
> Хороший вопрос. Вы можете создать воркеров Cloudflare бесплатно (100 тыс. Запросов в день), а затем вам нужно будет немного изменить библиотеку (см. [№ 62](https://github.com/jsanahuja/InstagramFeed/pull/62) ).
> Допустим, вы создали воркеров Cloudflare, вы можете использовать мою [модифицированную версию этой библиотеки](https://github.com/evaleries/InstagramFeed/blob/master/dist/InstagramFeed.min.js) . Затем вам нужно немного настроить экземпляр InstagramFeed, например
> 
> ```js
> новый  InstagramFeed ( { 
>     'username' : 'travisci' , 
>     'container' : document . getElementById ( "instagram-feed5" ) , 
>     'display_profile' : true , 
>     'cdnUrlResolver' : function ( url )  { 
>         return  'https: // your- cloudflare.workers.dev/ '  +  url ; 
>     } 
> } ) ;
> ```
> 
> Проверьте мою живую демонстрацию:
> [Основной сайт](https://pemro.id) - проверьте **последний** раздел **Instagram **
> [https://pemro.id/plugins/insta-feed/insta-feed.js ](https://pemro.id/plugins/insta-feed/insta-feed.js)
> https://pemro.id/plugins/insta-feed/InstagramFeed.min.js - Здесь используется моя [модифицированная версия этой библиотеки](https://github.com/evaleries/InstagramFeed/blob/master/dist/InstagramFeed.min.js)

Yes, but if I understand correctly, will I still need to migrate my domain's dns to cloudflare?

**Comment by evaleries** [comment]
> Yes, but if I understand correctly, will I still need to migrate my domain's dns to cloudflare?

No, You don't need to do that. Migrate your domain to Cloudflare is optional. But, If you need your cdn to be customized to your domain, then you need migrate your domain to Cloudflare (e.g: cdn.your-domain.com) as pointed No 1 on tutorial.

**Comment by A1exMoody** [comment]
> > Yes, but if I understand correctly, will I still need to migrate my domain's dns to cloudflare?
> 
> No, You don't need to do that. Migrate your domain to Cloudflare is optional. But, If you need your cdn to be customized to your domain, then you need migrate your domain to Cloudflare (e.g: cdn.your-domain.com) as pointed No 1 on tutorial.

awesome! Thx for ur solution )

**Comment by ribeiroeder** [comment]
@evaleries great solution! Could you share the configuration inserted in your cloudflare worker?

I tested it here just by creating one, and the default settings return me the 404 error.

**Comment by evaleries** [comment]
> @evaleries great solution! Could you share the configuration inserted in your cloudflare worker?
> 
> I tested it here just by creating one, and the default settings return me the 404 error.

@ribeiroeder First of all, I don't know your default settings. As far as I know if you copy the whole script on tutorial, you can't directly access the workers (It will says 404 Bad url). The workers require Instagram's post url. So the final example would be: https://your.workers.dev/Instagram_post_url
e.g [Check this link structure](https://ig-cdn.evaleries.workers.dev/https://scontent-sea1-1.cdninstagram.com/v/t51.2885-15/sh0.08/e35/c0.135.1080.1080a/s640x640/175903196_239059484670718_2036164081485783518_n.jpg?tp=1&_nc_ht=scontent-sea1-1.cdninstagram.com&_nc_cat=110&_nc_ohc=2d3QGyxGzKEAX8b6SPG&edm=ABfd0MgBAAAA&ccb=7-4&oh=899a5beab70a2b66621bcef22d2c0a11&oe=60B2A84C&_nc_sid=7bff83)

**Comment by ribeiroeder** [comment]
Yes, the script is assembling the url correctly, [see here](https://wild-hat-82ef.rbbo.workers.dev/https://scontent-atl3-2.cdninstagram.com/v/t51.2885-15/sh0.08/e35/s640x640/175578828_291926832466819_4219020923555088031_n.jpg?tp=1&_nc_ht=scontent-atl3-2.cdninstagram.com&_nc_cat=107&_nc_ohc=69i8F7vymicAX-27trD&edm=ABfd0MgBAAAA&ccb=7-4&oh=475222df9a53f917313dc9f2060ec286&oe=60B1F1BE&_nc_sid=7bff83)

Similar to your worker, but it seems to be missing some configuration within the cloudflare worker to not return the 404

**Comment by jsanahuja** [comment]
@evaleries @ribeiroeder you can directly use the `host` parameter provided by this library.
